org.eclipse.corrosion/src/org/eclipse/corrosion/CorrosionPreferencePage.java

```diff
@@ -382,34 +385,60 @@ private void createCommandPathsPart(Composite container) {
-				file.setExecutable(true);
-				String[] command = new String[] { "/bin/bash", "-c", file.getAbsolutePath() + " -y" }; //$NON-NLS-1$ //$NON-NLS-2$ //$NON-NLS-3$
-				ProcessBuilder builder = new ProcessBuilder(command);
-				Process process = builder.start();
-				if (process.waitFor() == 0) {
-					CorrosionPreferenceInitializer initializer = new CorrosionPreferenceInitializer();
-					initializer.initializeDefaultPreferences();
-					Display.getDefault().asyncExec(() -> {
-						setInstallRequired(false);
-						performDefaults();
-						setValid(isPageValid());
+		ProgressMonitorDialog dialog = new ProgressMonitorDialog(this.getShell());
+		try {
+			dialog.run(true, true, monitor -> {
+				monitor.beginTask(Messages.CorrosionPreferencePage_installingRustupCargo, 15);
+				try {
+					Bundle bundle = CorrosionPlugin.getDefault().getBundle();
+					URL fileURL = FileLocator.toFileURL(bundle.getEntry("scripts/rustup-init.sh")); //$NON-NLS-1$
+					File file = new File(new URI(fileURL.getProtocol(), fileURL.getPath(), null));
+					file.setExecutable(true);
+					String[] command = new String[] { "/bin/bash", "-c", file.getAbsolutePath() + " -y" }; //$NON-NLS-1$ //$NON-NLS-2$ //$NON-NLS-3$
+					Process process = new ProcessBuilder(command).start();
+					BufferedReader reader = new BufferedReader(new InputStreamReader(process.getErrorStream()));
+					CompletableFuture.runAsync(() -> {
+						reader.lines().forEachOrdered(line -> {
+							monitor.subTask(line);
+							monitor.worked(1);
+						});
 					});
-					RustManager.setDefaultToolchain("beta"); //$NON-NLS-1$
-					return;
+					while (process.isAlive() && !monitor.isCanceled()) {
+						Thread.sleep(50);
+					}
+					if (monitor.isCanceled()) {
+						process.destroy();
+						Display.getDefault().asyncExec(() -> {
+							setInstallRequired(true);
+						});
+					} else if (process.waitFor() == 0) {
+						CorrosionPreferenceInitializer initializer = new CorrosionPreferenceInitializer();
+						initializer.initializeDefaultPreferences();
+						Display.getDefault().asyncExec(() -> {
+							setInstallRequired(false);
+							performDefaults();
+							setValid(isPageValid());
+						});
+						RustManager.setDefaultToolchain("beta"); //$NON-NLS-1$
+					} else {
+						createFailedInstallPopup();
+					}
+				} catch (InterruptedException | URISyntaxException | IOException e) {
+					createFailedInstallPopup();
 				}
-			} catch (InterruptedException | URISyntaxException | IOException e) {
-				// will be caught with dialog
-			}
-			Display.getDefault().asyncExec(() -> {
-				setInstallRequired(true);
-				MessageDialog.openError(getShell(), Messages.CorrosionPreferencePage_cannotInstallRustupCargo, Messages.CorrosionPreferencePage_cannotInstallRustupCargo_details);
+				monitor.done();
 			});
-		}).schedule();
+		} catch (InvocationTargetException | InterruptedException e) {
+			createFailedInstallPopup();
+		}
+	}
+
+	private void createFailedInstallPopup() {
```

Comment:
It's not really "create", it's "show". Create in JFace is typically used to create an object and keep a reference to it, that's not the case here.